### PR TITLE
Increased test coverage

### PR DESCRIPTION
**Pull request by isseis**
This pull request adds comprehensive unit tests for several core modules, improving test coverage and reliability. The new tests target environment filtering, resource management, risk evaluation, and hash directory validation. These changes help ensure correct behavior under various scenarios and edge cases, and they utilize the `testify` library for clear assertions.

**Testing improvements:**

* Added extensive unit tests for environment variable filtering in `filter_test.go`, covering allowlists, parsing, filtering, and edge cases like empty values and special characters. (`internal/runner/environment/filter_test.go`)
* Introduced thorough tests for resource manager behavior, including dry-run and normal execution modes, cleanup operations, analysis recording, and debug info updates. (`internal/runner/resource/default_manager_test.go`)
* Rewrote and expanded risk evaluator tests to cover privilege escalation, destructive file operations, network/system modifications, safe commands, and risk level hierarchy, using the runtime command abstraction. (`internal/runner/risk/evaluator_test.go`)

**General test infrastructure:**

* Added `//go:build test` build tags and switched test package imports to use `testify` for improved assertions and test reliability. (`internal/cmdcommon/common_test.go`, `internal/runner/environment/filter_test.go`, `internal/runner/risk/evaluator_test.go`) [[1]](diffhunk://#diff-709a19d4b6185433fa3a9c278d527eec49049077f67d9c5fa55b8a95cb2eaa33R1-R76) [[2]](diffhunk://#diff-6e622a157cd1eb97a1502a3172439cb783f90e59af86ad63cfd04c4873bac0a4R4-R8) [[3]](diffhunk://#diff-1fa497682be1aa2ccbdfe9aa92fdb039005537a09a5d8e95f266ae6fe4942239L1-R295)
* Implemented unit tests for hash directory validator creation, handling valid, default, non-existent, relative, and empty paths, and verifying the default directory constant. (`internal/cmdcommon/common_test.go`)

**Documentation update:**

* Updated implementation plan documentation to reflect current test coverage percentage. (`docs/tasks/0055_command_output_redaction_for_slack/04_implementation_plan.md`)

### CODE REVIEW

File: internal/runner/environment/filter_test.go [hunk 1]
```diff
@@ -13,3 +15,198 @@ func TestNewFilter(t *testing.T) {
...
+	origEnv := os.Environ()
+	defer func() {
+		os.Clearenv()
+		for _, env := range origEnv {
+			key, value, ok := parseEnvString(env)
+			if ok {
+				os.Setenv(key, value)
+			}
+		}
+	}()
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/environment/filter_test.go]: done
